revocation/revocation.go

```diff
@@ -0,0 +1,122 @@
+// Package Revocation provides methods for checking the revocation status of a certificate chain
+package revocation
+
+import (
+	"crypto/x509"
+	"net/http"
+	"time"
+
+	"github.com/notaryproject/notation-core-go/revocation/ocsp"
+)
+
+// Revocation is an interface that specifies methods used for revocation checking
+type Revocation interface {
+	Validate(certChain []*x509.Certificate, signingTime time.Time) error
+	OCSPStatus(certChain []*x509.Certificate, signingTime time.Time) error
```

Comment:
1. Sounds good, will add in next commit.
2. My thought was that revocation would require both OCSPStatus and CRLStatus once we add CRL, since both are needed for it to be added. If this would be an issue for extensibility, we can just remove the wrapper around OCSPStatus and just call ocsp.OCSPStatus from the Validate function.